File: Expenses/Expenses.ApiRepository/Repositories/Households.cs
using Expenses.ApiRepository.Interfaces;
using Expenses.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace Expenses.ApiRepository.Repositories
{
    class Households : RepositoryBase, IHouseholds
    {
        public Households(string host, string path, string apiRepositoryName = "households") 
            : base(host, path, apiRepositoryName)
        {
        }

        public async Task<HttpResponseMessage> Invite(string invitersLogin, string invitedLogin, string key)
        {
            var uri = BaseUri.Append("invite", invitersLogin, invitedLogin);
            using (var httpClient = new HttpClient())
            {
                httpClient.DefaultRequestHeaders.Add("x-functions-key", key);
                return await httpClient.GetAsync(uri);
            }
        }

        public async Task<HttpResponseMessage> AcceptInvitation(string from, string to, string rowKey, string key)
        {
            var uri = BaseUri.Append("accept", from, to, rowKey);
            using (var httpClient = new HttpClient())
            {
                httpClient.DefaultRequestHeaders.Add("x-functions-key", key);
                return await httpClient.GetAsync(uri);
            }
        }

        public async Task<HttpResponseMessage> GetMembers(string householdId, string key)
        {
            var uri = BaseUri.Append("members", householdId);
            using (var httpClient = new HttpClient())
            {
                httpClient.DefaultRequestHeaders.Add("x-functions-key", key);
                return await httpClient.GetAsync(uri);
            }
        }
    }
}

File: Expenses.AndroidClient/Expenses.Repository/src/main/java/net/azurewebsites/expenses_by_klaudia/repository/Households.java
package net.azurewebsites.expenses_by_klaudia.repository;

import java.net.HttpURLConnection;

public class Households extends RepositoryBase {
    public Households(String host, String path) {
        super(host, path, "households");
    }

    public HttpURLConnection invite(String invitersLogin, String invitedLogin, String key) throws RepositoryException {
        String uri = GetBaseUriBuilder()
                .appendPath("invite")
                .appendPath(invitersLogin)
                .appendPath(invitedLogin)
                .appendQueryParameter("code", key)
                .build()
                .toString();
        return SendGet(uri);
    }

    public HttpURLConnection acceptInvitation(String from, String to, String rowkey, String key) throws RepositoryException {
        String uri = GetBaseUriBuilder()
                .appendPath("accept")
                .appendPath(from)
                .appendPath(to)
                .appendPath(rowkey)
                .appendQueryParameter("code", key)
                .build()
                .toString();
        return SendGet(uri);
    }

    public HttpURLConnection getMemebers(String householdId, String key) throws RepositoryException {
        String uri = GetBaseUriBuilder()
                .appendPath("members")
                .appendPath(householdId)
                .appendQueryParameter("code", key)
                .build()
                .toString();
        return SendGet(uri);
    }
}
